Sources/MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -184,7 +184,7 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
         return true
     }
     
-    public func userIsOnRoute(_ location: CLLocation) -> Bool {
+    public func userIsOnRoute(_ location: CLLocation) -> Bool? {
```

Comment:
We receive navigation statuses asynchronously, so it's possible that navigation status is still unknown.
Now we return `nil` in this case. This function is only used in the rerouting logic.

Reply:
Maybe we should just inline the `true` fallback here instead of scattering it in the call sites, where they could get inconsistent in the future? If we do have to make this change to the public Router protocol, let’s mention it in the changelog.